meson.build

```diff
@@ -143,9 +143,14 @@ configh_data.set('_CRT_NONSTDC_NO_DEPRECATE', 1)
 configh_data.set('WIN32_LEAN_AND_MEAN', 1)
 
 # Supports -Wl,--version-script?
+if cc.has_link_argument('-Wl,--undefined-version')
+    extra_linker_args = ',--undefined-version'
+else
+    extra_linker_args = ''
+endif
```

Comment:
Another option would be to assume it's only needed if `cc.get_linker_id() == 'ld.lld'`, which can be done as a string comparison instead of compiling a test program to check `has_link_argument`

Reply:
From a maintainer POV, it’s much more robust to use the builtin features of the build tool. And again, GNU ld does support this flag if recent enough.

Reply:
Whether GNU ld supports the flag is irrelevant -- what matters is whether GNU ld defaults to no-undefined-version, as it is the change in default that caused the original lld issue.

Which it currently doesn't. So handling this on GNU ld becomes a question of whether you:
- future-proof the code against a theoretical future change in the GNU binutils defaults (I personally am skeptical they will ever change, but that's a random personal opinion).
- simplify the meson.build support code and optimize the configure time by a small amount.

There are arguments to be made both ways. I did say it was another option, not that you must choose it.